com.unity.render-pipelines.universal/Tests/Runtime/Light2DTests.cs

```diff
@@ -96,5 +101,77 @@ public void LightIsNotInVisibleListIfNotInCameraView()
 
             Assert.IsFalse(cullResult.visibleLights.Contains(light));
         }
+
+        [Test]
+        public void CachedMeshDataIsUpdatedOnChange()
+        {
+            var shapePath = new Vector3[4] { new Vector3( 0, 0, 0), new Vector3(1,0,0), new Vector3(1, 1, 0), new Vector3(0, 1, 0) };
+            var cachedMesh = m_TestObjectCached.AddComponent<LightCachedMeshData>();
+            var light = m_TestObjectCached.AddComponent<Light2D>();
+            light.lightType = Light2D.LightType.Freeform;
+
+            cachedMesh.allowUpdateOnTests = true;
+            light.SetShapePath(shapePath);
+            light.UpdateMesh(true);
+
+            Assert.AreEqual(true, cachedMesh.HasGeometryData() );
+        }
+
+        [UnityTest]
+        public IEnumerator CachedMeshDataIsUsedOnRuntime()
+        {
+            var shapePath = new Vector3[4] { new Vector3( 0, 0, 0), new Vector3(1,0,0), new Vector3(1, 1, 0), new Vector3(0, 1, 0) };
+            var cachedMesh = m_TestObjectCached.AddComponent<LightCachedMeshData>();
+            var light = m_TestObjectCached.AddComponent<Light2D>();
+            light.lightType = Light2D.LightType.Freeform;
+            cachedMesh.allowUpdateOnTests = true;
+            light.SetShapePath(shapePath);
+            light.UpdateMesh(true);
+
+            yield return new EnterPlayMode();
+            yield return null;
+
+            // Check if Cached Data and the actual data are the same.
+            Assert.AreEqual(true, cachedMesh.HasGeometryData());
+            Assert.AreEqual(light.lightMesh.triangles.Length, cachedMesh.indices.Length );
+            Assert.AreEqual(light.lightMesh.vertices.Length, cachedMesh.vertices.Length );
+
+            yield return new ExitPlayMode();
+        }
+
+        [UnityTest]
+        public IEnumerator CachedMeshDataIsOverriddenByRuntimeChanges()
+        {
+            var shapePath = new Vector3[4] { new Vector3( 0, 0, 0), new Vector3(1,0,0), new Vector3(1, 1, 0), new Vector3(0, 1, 0) };
+            var cachedMesh = m_TestObjectCached.AddComponent<LightCachedMeshData>();
+            var light = m_TestObjectCached.AddComponent<Light2D>();
+            light.lightType = Light2D.LightType.Freeform;
+            cachedMesh.allowUpdateOnTests = true;
+            light.SetShapePath(shapePath);
+            light.UpdateMesh(true);
+
+            yield return new EnterPlayMode();
+            yield return null;
+
+            // Check if Cached Data and the actual data are the same.
+            Assert.AreEqual(true, cachedMesh.HasGeometryData());
+            Assert.AreEqual(light.lightMesh.triangles.Length, cachedMesh.indices.Length );
+            Assert.AreEqual(light.lightMesh.vertices.Length, cachedMesh.vertices.Length );
```

Comment:
These are already covered by the previous test, no?

Reply:
1) CachedMeshDataIsUpdatedOnChange -> Checks if GeometryCache is updated on Editor.
2) CachedMeshDataIsUsedOnRuntime      -> Checks if GeometryCache is used on Playmode if one Exists.
3) CachedMeshDataIsOverriddenByRuntimeChanges -> Checks if GeometryCache is overridden if there is change in Shape.